packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -160,200 +157,101 @@ class DefaultTextEditingShortcuts extends Shortcuts {
-    //   * Meta + V
-    //   * Meta + A
-    //   * Meta + arrow down
-    //   * Meta + arrow left
-    //   * Meta + arrow right
-    //   * Meta + arrow up
-    //   * Meta + shift + arrow down
-    //   * Meta + shift + arrow left
-    //   * Meta + shift + arrow right
-    //   * Meta + shift + arrow up
-    //   * Shift + end
-    //   * Shift + home
-    //   * Meta + delete
-    //   * Meta + backspace
+    // Shift + Arrow: Extend Selection.
+    SingleActivator(LogicalKeyboardKey.arrowLeft, shift: true): ExtendSelectionByCharacterIntent(forward: false, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowRight, shift: true): ExtendSelectionByCharacterIntent(forward: true, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowUp, shift: true): ExtendSelectionVerticallyToAdjacentLineIntent(forward: false, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowDown, shift: true): ExtendSelectionVerticallyToAdjacentLineIntent(forward: true, collapseSelection: false),
+
+    SingleActivator(LogicalKeyboardKey.arrowLeft, alt: true): ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+    SingleActivator(LogicalKeyboardKey.arrowRight, alt: true): ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+    SingleActivator(LogicalKeyboardKey.arrowUp, alt: true): ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: true),
+    SingleActivator(LogicalKeyboardKey.arrowDown, alt: true): ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: true),
+
+    SingleActivator(LogicalKeyboardKey.arrowLeft, shift: true, alt: true): ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowRight, shift: true, alt: true): ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowUp, shift: true, alt: true): ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowDown, shift: true, alt: true): ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: false),
+
+    SingleActivator(LogicalKeyboardKey.arrowLeft, control: true): ExtendSelectionToNextWordBoundaryIntent(forward: false, collapseSelection: true),
+    SingleActivator(LogicalKeyboardKey.arrowRight, control: true): ExtendSelectionToNextWordBoundaryIntent(forward: true, collapseSelection: true),
+
+    SingleActivator(LogicalKeyboardKey.arrowLeft, shift: true, control: true): ExtendSelectionToNextWordBoundaryIntent(forward: false, collapseSelection: false),
+    SingleActivator(LogicalKeyboardKey.arrowRight, shift: true, control: true): ExtendSelectionToNextWordBoundaryIntent(forward: true, collapseSelection: false),
+
+    SingleActivator(LogicalKeyboardKey.keyX, control: true): CopySelectionTextIntent.cut(SelectionChangedCause.keyboard),
```

Comment:
These shortcuts aren't used for Mac due to `control` vs. `meta`, right?  I expected _commonShortcuts to be used for all platforms at first before I thought of this.

Reply:
No unfortunately. I'll put a comment there. I'm a bit curious how macOS handles a keyboard layout that doesn't have a command key. Does it automatically detect that and change the shortcuts?

Reply:
Oh for macOS we probably just have to translate `moveToRightEndOfLine` to the right intent and fire the intent from the primary focus according to #30725.